docs/docs/api/optimizers/GEPA/GEPA_Advanced.md

```diff
@@ -0,0 +1,435 @@
+# dspy.GEPA - Advanced Features
+
+## Custom Instruction Proposers
+
+### What is instruction_proposer?
+
+The `instruction_proposer` is the component responsible for invoking the `reflection_lm` and proposing new prompts during GEPA optimization. When GEPA identifies underperforming components in your DSPy program, the instruction proposer analyzes execution traces, feedback, and failures to generate improved instructions tailored to the observed issues.
+
+### Default Implementation
+
+By default, GEPA uses the built-in instruction proposer from the [GEPA library](https://github.com/gepa-ai/gepa), which implements the [`ProposalFn`](https://github.com/gepa-ai/gepa/blob/main/src/gepa/core/adapter.py). The default proposer uses this prompt template:
```

Comment:
Link to the default proposer as well https://github.com/gepa-ai/gepa/blob/main/src/gepa/proposer/reflective_mutation/reflective_mutation.py#L53-L75